lib/HttpSignatureStrategy.js

```diff
@@ -229,7 +245,7 @@ function getHttpsPublicKey(publicKey, callback) {
   async.auto({
     dereferencePublicKey: function(callback) {
       jsonRequest.get(publicKey, function(err, res) {
-        if(err) {
+        if(err || res.statusCode !== 200) {
```

Comment:
Since we added this extra check, we probably want the `httpStatusCode` reported in the error to be `400` if `err` is truthy and `res.statusCode` if not. Right now it's just always `400`.